sql-plugin/src/main/spark320/scala/com/nvidia/spark/rapids/shims/HybridFileSourceScanExecMeta.scala

```diff
@@ -0,0 +1,188 @@
+  // if possible. Instead regarding filters as childExprs of current Meta, we create
+  // a new meta for SubqueryBroadcastExec. The reason is that the native replacement of
+  // FileSourceScan is independent from the replacement of the partitionFilters.
+  private lazy val partitionFilters = {
+    val convertBroadcast = (bc: SubqueryBroadcastExec) => {
+      val meta = GpuOverrides.wrapAndTagPlan(bc, conf)
+      meta.tagForExplain()
+      meta.convertIfNeeded().asInstanceOf[BaseSubqueryExec]
+    }
+    wrapped.partitionFilters.map { filter =>
+      filter.transformDown {
+        case dpe@DynamicPruningExpression(inSub: InSubqueryExec) =>
+          inSub.plan match {
+            case bc: SubqueryBroadcastExec =>
+              dpe.copy(inSub.copy(plan = convertBroadcast(bc)))
+            case reuse@ReusedSubqueryExec(bc: SubqueryBroadcastExec) =>
+              dpe.copy(inSub.copy(plan = reuse.copy(convertBroadcast(bc))))
+            case _ =>
+              dpe
+          }
+      }
+    }
+  }
+
+  // partition filters and data filters are not run on the GPU
+  override val childExprs: Seq[ExprMeta[_]] = Seq.empty
+
+  override def tagPlanForGpu(): Unit = {
+    val cls = wrapped.relation.fileFormat.getClass
+    if (cls != classOf[ParquetFileFormat]) {
+      willNotWorkOnGpu(s"unsupported file format: ${cls.getCanonicalName}")
+    }
+  }
+
+  override def convertToGpu(): GpuExec = {
+    // Modifies the original plan to support DPP
+    val fixedExec = wrapped.copy(partitionFilters = partitionFilters)
+    HybridFileSourceScanExec(fixedExec)(conf)
+  }
+}
+
+object HybridFileSourceScanExecMeta {
+  // Determines whether using HybridScan or GpuScan
+  def useHybridScan(conf: RapidsConf, fsse: FileSourceScanExec): Boolean = {
+    val isEnabled = if (conf.useHybridParquetReader) {
+      require(conf.loadHybridBackend,
+        "Hybrid backend was NOT loaded during the launch of spark-rapids plugin")
+      true
+    } else {
+      false
+    }
+    // Currently, only support reading Parquet
+    lazy val isParquet = fsse.relation.fileFormat.getClass == classOf[ParquetFileFormat]
+    // Check if data types of all fields are supported by HybridParquetReader
+    lazy val allSupportedTypes = !fsse.requiredSchema.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, {
+        // Currently, under some circumstance, the native backend may return incorrect results
+        // over MapType nested by nested types. To guarantee the correctness, disable this pattern
+        // entirely.
+        // TODO: figure out the root cause and support it
+        case ArrayType(_: MapType, _) => true
+        case MapType(_: MapType, _, _) | MapType(_, _: MapType, _) => true
+        case st: StructType if st.exists(_.dataType.isInstanceOf[MapType]) => true
+        // TODO: support DECIMAL with negative scale
+        case dt: DecimalType if dt.scale < 0 => true
+        // TODO: support DECIMAL128
```

Comment:
nit: again having an issue to point to is helpful

Reply:
https://github.com/NVIDIA/spark-rapids/issues/11966